xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java

```diff
@@ -4534,7 +4534,7 @@ private XWikiDocument cloneInternal(DocumentReference newDocumentReference,
             if (cloneArchive) {
                 doc.cloneDocumentArchive(this);
             } else {
-                doc.setDocumentArchive(getDocumentArchive());
+                doc.setDocumentArchive(new XWikiDocumentArchive(newDocumentReference.getWikiReference(), getId()));
```

Comment:
Would be probably better to add a comment on this one, or we'll have a bit of WTF when reading it in few weeks.

Reply:
Indeed, documentation added.